src/python/lib/ensembl/compara/utils/taxonomy/taxon_selector.py

```diff
@@ -0,0 +1,101 @@
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Collection of taxonomy methods to configure and extract taxonomy related information.
+
+Typical usage examples::
+
+    from ensembl.compara.utils.taxonomy import *
+
+    with self.dbc.session_scope() as session:
+        ref_dirs = collect_taxonomys_from_path(session, self.dir)
+        ref_taxon = match_taxon_to_reference(session, taxon_name, ref_dirs)
+
+"""
+
+import os
+from pathlib import Path
+from typing import List, Union
+
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import NoResultFound
+
+from ensembl.ncbi_taxonomy.api.utils import Taxonomy
+from ensembl.ncbi_taxonomy.models import NCBITaxaName
+
+
+def collect_taxonomys_from_path(session: Session, rootdir: Union[str, Path]) -> list:
+    """Returns a list of ncbi_taxa_name.names list(str)
+
+    Args:
+        rootdir: collections directory containing directories
+        named by taxonomic classification
+    """
+    subdir_names = [f.name for f in os.scandir(rootdir) if f.is_dir(follow_symlinks=True)]
+    valid_taxons = [filter_real_taxon(session, name) for name in subdir_names]
+    taxon_list = list(filter(None, valid_taxons)) # type: List[str]
+    return taxon_list
+
+
+def match_taxon_to_reference(
+    session: Session, taxon_name: str, taxon_list: list
+) -> str:
+    """Returns a taxonomic clade name within the ``taxon_list`` and ``taxon_name`` ancestry
+
+    Args:
+        taxon_name: Scientific ncbi_taxa_name.name of genome in database
+        taxon_list: List of clade ncbi_taxa_name.names
+    """
+
+    node = Taxonomy.fetch_taxon_by_species_name(session, taxon_name)
+    ancestor_nodes = Taxonomy.fetch_ancestors(session, node.taxon_id)
+    ancestors = [ancestor["taxon_id"] for ancestor in ancestor_nodes]
+    ancestors.reverse()
+    for ancestor in ancestors:
+        ancestor_name = fetch_scientific_name(session, ancestor)
+        if any(str(ancestor_name).lower() in taxon.lower() for taxon in taxon_list):
+            return str(ancestor_name).lower()
+    return "default"
+
+
+def filter_real_taxon(session: Session, taxon: str):
+    """Returns a taxon name if it exists
+
+    Args:
+        taxon_name: Scientific ncbi_taxa_name.name of genome in database
+    """
+    try:
+        Taxonomy.fetch_taxon_by_species_name(session, taxon)
+        return taxon
+    except NoResultFound:
+        return None
```

Comment:
I believe (not sure) PEP recommendations for this would
```
except NoResultFound:
pass

return None
```

Reply:
I had to change this to be this way because the pylint/mypy complained when I was initially did it this way :shrug: